distribution/lib/Standard/Table/0.0.0-dev/src/Data/Table.enso

```diff
@@ -726,6 +729,85 @@ type Table
+         If `matcher` is a `Regex_Matcher`, `new_text` can include replacement
+         patterns (such as `$<n>`) for a marked group.
+       - mode: Specifies which instances of term the engine tries to find. When the
+         mode is `First` or `Last`, this method replaces the first or last instance
+         of term in each individual table cell. If set to `All`, it replaces all
+         instances of term.
+       - matcher: If a `Text_Matcher`, the text is compared using case-sensitivity
+         rules specified in the matcher. If a `Regex_Matcher`, the term is used as a
+         regular expression and matched using the associated options.
+       - on_problems: Specifies how to handle if a problem occurs, raising as a
+         warning by default.
+
+         The following problems can occur:
+         - If a column in columns is not in the input table, a `Missing_Input_Columns`.
+         - If duplicate columns, names or indices are provided, a
+           `Duplicate_Column_Selectors`.
+         - If a column index is out of range, a `Column_Indexes_Out_Of_Range`.
+         - If two distinct indices refer to the same column, an
+           `Input_Indices_Already_Matched`, with the column included the first
+           time it is matched.
+         - If a column in columns does not have a storage type of `Text`, or `Any`,
+           thus it is guaranteed that it can't contain any text values, a
+           `Invalid_Value_Type`.
+
+       > Example
+         Replace dashes with underscores in a column named "variable_names".
+
+             table.replace_text "variable_names" "-" "_" 
+
+       > Example
+         Remove leading and trailing spaces from cells in multiple columns.
+
+             table.replace_text (Column_Selector.By_Name ["foo", "bar"]) "^\s*(.*?)\s*$" "$1" matcher=Regex_Matcher.Regex_Matcher_Data
+
+       > Example
+         Replace texts in quotes with parentheses in column at index 1.
+
+             table.replace_text 1 '"(.*?)"' '($1)' matcher=Regex_Matcher.Regex_Matcher_Data
+    replace_text : (Text | Integer | Column_Selector) -> Text -> Text -> Matching_Mode | Regex_Mode -> (Text_Matcher | Regex_Matcher) -> Problem_Behavior -> Table
+    replace_text self columns=(Column_Selector.By_Index [0]) term="" new_text="" mode=Regex_Mode.All matcher=Text_Matcher.Case_Sensitive on_problems=Problem_Behavior.Report_Warning = if term.is_empty then self else
+        problem_builder = Problem_Builder.new
+        
+        selector = case columns of
+            _ : Column_Selector -> columns
+            name : Text -> Column_Selector.By_Name [name]
+            index : Integer -> Column_Selector.By_Index [index]
+        selection = Table_Helpers.select_columns_helper self.columns selector reorder=False problem_builder
+        selected_names = Map.from_vector (selection.map column-> [column.name, True])
+
+        map_preserve_name column f = column.map f . rename column.name
```

Comment:
I'm wondering if we should just add a `preserve_name:Boolean = False` flag to the `Column.map` operation instead? Sounds like it could be useful in other places too.

Reply:
lets leave for now - will add a `name` argument in the derived column work.

Reply:
It seems that this is very easy to do (either to add `preserve_name` or `name`), but that would be a public api change. Is adding a new argument with default backward compatible, or is it a breaking change?

If we want this option in the API anyway, then I don't mind changing it here. Up to you :)

Reply:
I think in general it should be backwards compatible.

I'm not sure if there are maybe some rare cases when it's not. But in that case - I'm convinced that just adding this new arg with a default should not break any existing code/tests.

Reply:
Because there are multiple options what the extra argument should be, let's leave it for a separate task. The only consequence here is that we have that extra `.rename column.name` in expression, which is not a big deal anyway.